### PR TITLE
Removes obsolete check in update_revive (thus fixing SR revival of people dead to tox)

### PR DESCRIPTION
**Pull request by moxian**
## What Does This PR Do
Allows one to successfully use strange reagent on dead people with more than 200 tox damage on them.

## Why It's Good For The Game
Currently, if you have a dead person with more than 200 toxins damage on them, you can't revive them via SR. If you try to - nothing happens, the person is not revived, their death timer is not reset. This is weird, because:
- you can't purge toxins damage out of dead people
- SR does check for the kinds of damage it cares about (brute, burn) and gibs the patient if it doesn't like them
- the chat message clearly states "Steve Nanotrasen seems to rise from the dead!"
This latter one especially indicates that this is more of a bugfix.

See (behaviour on live, note the "undefibbable" skull on medhud):  
image

## Technical details
All the callsites of `update_reviving` proc already check for target's health, and only ever try to revive when the health is above the "good" threshold (latter depends on the kind of reviving performed). So this health check in `can_be_revived` is not adding any value.

## Changelog
:cl:
fix: SR now works on people with > 200 tox damage
/:cl:

### CODE REVIEW

**Comment by marlyn-x86** [comment]
What does this do to IPCs

**Comment by marlyn-x86** [comment]
Tbh I'd recommend instead doing [this](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/objects/items/weapons/defib.dm#L459-L464) like the defib does instead of tweaking revive check code

File: code/modules/mob/living/stat_states.dm [hunk 1]
```diff
@@ -35,9 +35,6 @@
 
 /mob/living/proc/can_be_revived()
 	. = TRUE
-	// if(health <= min_health)
-	if(health <= HEALTH_THRESHOLD_DEAD)
-		return FALSE
```
Comment (marlyn-x86) [inline code/modules/mob/living/stat_states.dm:40]: Use the [defib code](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/objects/items/weapons/defib.dm#L459-L464) approach in SR's touch application method instead of doing this

**Review by marlyn-x86 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/stat_states.dm (hunk 1 above)
Comment (moxian) [reply code/modules/mob/living/stat_states.dm:40]: I don't think it's a good idea. Please see the comment below (above?) for my reasoning.

**Comment by marlyn-x86** [comment]
The defib's healing code is honestly pretty wonky, yeah, if you double-dip on multiple damage types, beyond a certain point you can get way more healing than necessary to get to the death threshold (which at this point is also somewhat of an outmoded concept anyhow), that being what the can_be_revived check was centered around in the first place (to avoid autodeaths and other strange behavior - it was meant to bring a body already in the condition to be alive back to life, a final step of bookkeeping once everything was back in order)

(if we were to keep the defib healing stuff in, the right way to do it would be to instead heal each individual damage type by the proportion of how much total damage the patient had already taken, so that the remaining damage after healing would sum together just above the death threshold. the logic for that would be a bit more complex, but tbh as-is the defib heal logic is already kind of hard to follow - without the min statement there, for example, patients sufficiently above the threshold would turn into soup upon being defibrillated as they took calamitous damage of every type)

The drone revival system also appears to be dummied out at this point, and the emag check should be able to just be done before any relevant revive calls are done, should we ever re-add that feature.

With your survey of all possible sources of update_revive, I deem it acceptable to amputate the can_be_revived check wholesale and instead move any necessary prep checks for the revive to around where update_revive is called. (no such movements actually need to be made at the moment, as per your survey, but future contributions involving this proc will now require this)

I'd advocate to rename the PR to "removes obsolete check in update_revive" while keeping the CL entry, I think what's mostly setting off my alarm bells is implementing a fix for a specific feature as a modification of a code path used by many other features
